drivers/volume/portworx/portworx.go

```diff
@@ -1074,7 +1099,7 @@ func (p *portworx) GetVolumeSnapshotRestoreStatus(snapRestore *stork_crd.VolumeS
 		return nil
 	}
 	for _, volInfo := range snapRestore.Status.Volumes {
-		taskID := restoreTaskPrefix + volInfo.Snapshot + "-" + volInfo.Volume
+		taskID := restoreTaskPrefix + volInfo.Snapshot + "-" + volInfo.Volume + string(snapRestore.GetUID())
```

Comment:
The return from `checkAndUpdateHaLevel` still needs to be fixed. If it returns err we shouldn't set the status to Failed. It should return err so that the controller can retry.